src/graph/query_graph.c

```diff
@@ -211,10 +211,6 @@ QueryGraph* QueryGraph_New() {
     g->edge_count = 0;
     g->node_cap = DEFAULT_GRAPH_CAP;
     g->edge_cap = DEFAULT_GRAPH_CAP;
-    g->nodes = (Node**)malloc(sizeof(Node*) * g->node_cap);
-    g->edges = (Edge**)malloc(sizeof(Edge*) * g->edge_cap);
-    g->node_aliases = (char**)malloc(sizeof(char*) * g->node_cap);
-    g->edge_aliases = (char**)malloc(sizeof(char*) * g->edge_cap);
```

Comment:
I think it would be better to have a single way of creating a querygraph object, and so consider merging the two into a single function, accepting both node_cap and edge_cap.

Reply:
That definitely makes more sense! Updated.